### ISSUE TITLE
Update the fsfinder to use concurrency

### ISSUE THREAD

**Issue by kehoecj** [Boeing/config-file-validator#61]
The current implementation of the file system finder is suitable for most use-cases but can get slow when recursing through a very large directory tree. It would improve performance to update the fsfinder to launch a new goroutine for each subdirectory. Determine a benchmark and determine if the worker pool needs to be limited to maximize performance as well as measuring the performance increase using goroutines

**Comment by AshwinKul28** [comment]
I can pick this up! @kehoecj

**Comment by kehoecj** [comment]
It's yours!

**Comment by kehoecj** [comment]
@AshwinKul28 Are you still interested in working on this issue?

**Comment by Axecalever** [comment]
@kehoecj, can I take this ?

**Comment by kehoecj** [comment]
> @kehoecj, can I take this ? 

Absolutely! I assigned it to you

**Comment by Axecalever** [comment]
@kehoecj  is there any specific dataset of config files that I could benchmark against?

**Comment by kehoecj** [comment]
> @kehoecj is there any specific dataset of config files that I could benchmark against?

No, we don't have a current dataset to benchmark against. It would be less about a dataset and more about a large tree of subdirectories, for instance running the validator against `/etc`, `/`, etc

**Comment by kehoecj** [comment]
@Axecalever Did you still want to work this issue?

**Comment by nikonhub** [comment]
Hello.

What do you think about using an external module like https://github.com/iafan/cwalk
I've tried it, and it seems 2x times faster.

**Comment by kehoecj** [comment]
@nikonhub You're welcome to put up a draft MR with the implementation
